test/functional/test_framework/test_node.py

```diff
@@ -151,6 +164,39 @@ def node_encrypt_wallet(self, passphrase):
         self.encryptwallet(passphrase)
         self.wait_until_stopped()
 
+    def add_p2p_connection(self, p2p_conn_type, dstaddr='127.0.0.1', dstport=None, services=NODE_NETWORK, send_version=True):
+        """Add a p2p connection to the node.
+
+        This method adds the p2p connection to the self.p2ps list and also
+        returns the connection to the caller."""
+        if dstport is None:
+            dstport = p2p_port(self.index)
+        p2p_conn = p2p_conn_type()
+        self.p2ps.append(p2p_conn)
+        p2p_conn.add_connection(NodeConn(dstaddr, dstport, self.rpc, p2p_conn, services=services, send_version=send_version))
+
+        return p2p_conn
+
+    @property
+    def p2p(self):
+        """Return the first p2p connection
+
+        Convenience property - most tests only use a single p2p connection to each
+        node, so this saves having to write node.p2ps[0] many times."""
+        assert self.p2ps, "No p2p connection"
+        return self.p2ps[0]
+
+    def send_message(self, message):
```

Comment:
Maybe call it `send_p2p_message` to be clearer and consistent with `add_p2p_connection`. This class has a lot of functionality already so it would be nice if all the new p2p attributes mentioned p2p.

Another alternative would be to drop this since `node.send_p2p_message()` hardly seems better than `node.p2p.send_message()`